src/main/java/seedu/address/logic/commands/schedule/EditEventCommand.java

```diff
@@ -93,15 +94,18 @@ public CommandResult execute(Model model) throws CommandException {
      * Creates and returns a {@code Schedule} with the details of {@code scheduleToEdit}
      * edited with {@code editEventDescriptor} at {@code targetEventIndex}.
      */
-    private static Schedule createEditedSchedule(
-            Schedule scheduleToEdit, Index targetEventIndex, EditEventDescriptor editEventDescriptor) {
+    private static Schedule createEditedSchedule(Schedule scheduleToEdit, Index targetEventIndex,
+            EditEventDescriptor editEventDescriptor) throws CommandException {
         assert scheduleToEdit != null;
 
         List<Event> scheduleEvents = scheduleToEdit.getEvents();
         ArrayList<Event> updatedEvents = new ArrayList<>(scheduleEvents);
 
         Event toEditEvent = updatedEvents.remove(targetEventIndex.getZeroBased());
         Event updatedEvent = createEditedEvent(toEditEvent, editEventDescriptor);
+        if (!updatedEvent.isValidDurationWithRecurFrequency()) {
```

Comment:
I think this would be better thrown in createEditedEvent? The event should not be created if the values were wrong

Reply:
The event has to be created for this to happen. It should either be done before creating the `Event` or during the creation of the `Event`. I have tried to consider both cases. But did not manage to find a good way to solve it. Thus, I resort to checking after the creation of the `Event`.
Why it does not work if you do this before creating the `Event`:
1. You would have to make sure that you do this check for every command that would change the event.
2. If we were to do this check in the Parser commands, it would be awkward, since this class is meant for just retrieving the input. 
3. If we were to do this in the Command classes, retrieving the information of events here would also be very awkward, since this class is meant for the execution of the command.

Why it does not work if you do this during creation of the `Event`:
1. It actually works, however, we would not be able to throw the exception message if we were to check in the constructor. It will also require us to use try catch in every time we create an event.
2. There is a workaround the constructor problem, which is to have a static `of` method. However, personally, I think if we were to have a this static method, then we should make the constructor private to prevent misuse of the `Event`. But it would be a hassle to have to change many of the implementations we already have in place for our scheduling. 

Let me know if would still like the change.

Reply:
Hmm, how about we do the checks in `Event#isValidEvent`, and in `createEditedEvent` we throw the exception if not valid? Otherwise we can go with this I guess.

Reply:
I think for this one can move the checking to `createEditedEvent` and you don't even need to catch it in createEditedSchedule because it will throw a CommandException anyway.